framework/src/base/FEProblem.C

```diff
@@ -500,6 +500,51 @@ void FEProblem::initialSetup()
        ++it)
     it->second->updateSeeds(EXEC_INITIAL);
 
+  if (!_app.isRecovering())
+  {
+    Moose::setup_perf_log.push("Initial computeUserObjects()","Setup");
+
+    computeUserObjects(EXEC_INITIAL, UserObjectWarehouse::PRE_AUX);
+
+    _aux.compute(EXEC_INITIAL);
```

Comment:
It seems wrong that we are doing any execution before `initialSetup()` has been run on all systems. The way it is now, all systems are initialized, then executed. In fact, I was really thinking about breaking `initialSetup()` and `executeInitial()` into two different methods to ease with the setup of SubApps. I think you are running into the same problem I was: When you running initialSetup() on the MultiApps, you are also doing some amount of execution.

I suggest that we break these methods into two routines so that we have even more control over these startup tasks.